ic-agent/src/agent/mod.rs

```diff
@@ -136,9 +151,25 @@ impl Agent {
             ingress_expiry_duration: config
                 .ingress_expiry_duration
                 .unwrap_or_else(|| Duration::from_secs(300)),
+            root_key: RwLock::new(DEFAULT_ROOT_KEY.to_vec()),
         })
     }
 
+    /// Fetch the root key from the status endpoint.
+    /// It is not necessary to call this when communicating with "the" Internet Computer.
+    pub async fn fetch_root_key(&self) -> Result<(), AgentError> {
+        let status = self.status().await?;
+        let root_key = status.root_key.ok_or_else(AgentError::NoRootKeyInStatus)?;
+        let mut write_guard = self.root_key.write().unwrap();
+        *write_guard = root_key;
```

Comment:
```suggestion
        if let Ok(mut write_guard) = self.root_key.write() {
            *write_guard = root_key;
        }
```

Reply:
getting rid of the unwraps

Reply:
Ok, I applied these changes.  The reason I had them as `unwrap()` is because the RwLock poisoning condition that would cause them to fail can't happen in the code as is, because the writer can't panic.